GNOME: 48 → 49

jtojnar opened the pull request:
This draft PR starts update work for the upcoming GNOME 49 release.

- Schedule: https://release.gnome.org/calendar/ Release: 2025-09-13
- Project: https://github.com/orgs/NixOS/projects/101
- Previous major releases: https://github.com/NixOS/nixpkgs/pull/386514
- Tool for reviewing changes: https://github.com/jtojnar/nonemast
- Release announcement: <!-- https://release.gnome.org/49/ -->
- Hydra jobset: TBD <!-- https://hydra.nixos.org/jobset/nixpkgs/gnome -->

### Major changes

- UNIX specific Gio API moved to GioUnix https://gitlab.gnome.org/GNOME/glib/-/issues/3744
- GI bindings might use wrong getter https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/529
- pygobject3 switched to libgirepository-2.0, there are downstream reports of some libpeas plugins breaking https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1099164
- gdk-pixbuf disabled thumbnailer by default https://github.com/GNOME/gdk-pixbuf/commit/e110b0b23995da73990130c480f5b6b240e78a26
- gdm switched to dynamic users for greeter sessions https://gitlab.gnome.org/GNOME/gdm/-/merge_requests/289

### How to help

TBD

<!-- 
- test running your system from this branch (warning: you might not be able to roll back to 45 without dropping all configuration)
- test running new versions of software from this branch
- check hydra builds
  1. look at the latest evaluation in the [jobset](https://hydra.nixos.org/jobset/nixpkgs/gnome)
  2. check the _newly/still failing jobs_ tab, look for red crosses ❌ 
  3. open the individual failures' logs
  4. if you see any failures that look like dependency issue or anything that could be caused by this PR, report it in a comment, or open a PR against gnome-46 branch, if you are able to.
- report any and all problems!
-->

### Depends on

https://github.com/NixOS/nixpkgs/pull/422662

jtojnar committed:
gnome-keysign: Fix eval

future package is marked as broken so this was not evaluating.
As a result, in conjunction with https://github.com/python/cpython/issues/103847,
update.nix froze when trying to bump GNOME package set.

jtojnar committed:
gnome-recipes: Remove

Unmaintained upstream and depends on libsoup 2 which is insecure.

jtojnar committed:
maintainers/scripts/update.nix: Print stderr on instantiation failure

For some reason `asyncio.create_subprocess_exec` freezes after failing previous call returns `CalledProcessError`.
Perhaps because `asyncio.gather` fails and cancels the later tasks, triggering https://github.com/python/cpython/issues/103847

jtojnar committed:
gssdp: Unmaintain

It is long deprecated and unmaintained upstream and depends on insecure libraries.

jtojnar committed:
libchamplain: Unmaintain

Only maintain libsoup3 based version.
Eval failure was breaking GNOME updates.

Also change the canonical attribute path for `updateScript` to the maintained variant.

jtojnar committed:
libsoup: Unmaintain

It is abandoned upstream, projects should port to libsoup3.

jtojnar committed:
gnome-initial-setup: 48.1 → 49.beta.1

https://gitlab.gnome.org/GNOME/gnome-initial-setup/-/compare/48.1...49.beta.1

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>
Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
epiphany: 48.5 → 49.rc

https://gitlab.gnome.org/GNOME/epiphany/-/compare/48.5...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>
Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-logs: 45.0 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-logs/-/compare/gnome-logs-45.0...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-connections: 48.0 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-connections/-/compare/48.0...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>
Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
xdg-desktop-portal-gnome: 48.0 → 49.rc

https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/compare/48.0...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-chess: 48.0 → 49.0

https://gitlab.gnome.org/GNOME/gnome-chess/-/compare/48.0...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-sudoku: 48.1 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-sudoku/-/compare/48.1...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
ghex: 48.0 → 48.1

https://gitlab.gnome.org/GNOME/ghex/-/compare/48.0...48.1

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
quadrapassel: 40.2 → 49.rc

https://gitlab.gnome.org/GNOME/quadrapassel/-/compare/40.2...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
loupe: 48.1 → 49.rc

https://gitlab.gnome.org/GNOME/loupe/-/compare/48.1...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-mahjongg: 48.1 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-mahjongg/-/compare/48.1...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-font-viewer: 48.0 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-font-viewer/-/compare/48.0...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-builder: 48.2 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-builder/-/compare/48.2...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-mines: 48.1 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-mines/-/compare/48.1...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>
Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-software: 48.4 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-software/-/compare/48.4...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
snapshot: 48.0.1 → 49.rc

https://gitlab.gnome.org/GNOME/snapshot/-/compare/48.0.1...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>
Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
dconf-editor: 45.0.1 → 49.rc

https://gitlab.gnome.org/GNOME/dconf-editor/-/compare/45.0.1...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>
Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
simple-scan: 48.1 → 49.0

https://gitlab.gnome.org/GNOME/simple-scan/-/compare/48.1...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-session: 48.0 → 49.rc

X11 support now disabled by default.

https://gitlab.gnome.org/GNOME/gnome-session/-/compare/48.0...49.rc

Co-authored-by: Bobby Rong <rjl931189261@126.com>
Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
metacity: 3.56.0 → 3.57.1

https://gitlab.gnome.org/GNOME/metacity/-/compare/3.56.0...3.57.1

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-system-monitor: 48.1 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-system-monitor/-/compare/48.1...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>
Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-maps: 48.6 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-maps/-/compare/48.6...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-calculator: 48.1 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-calculator/-/compare/48.1...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>
Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-nibbles: 4.2.2 → 4.4.rc2

https://gitlab.gnome.org/GNOME/gnome-nibbles/-/compare/4.2.2...4.4.rc2

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>
Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-control-center: 48.4 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-control-center/-/compare/48.4...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>
Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-weather: 48.0 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-weather/-/compare/48.0...49.rc

Co-authored-by: Bobby Rong <rjl931189261@126.com>
Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-music: 48.1 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-music/-/compare/48.1...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-characters: 48.0 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-characters/-/compare/48.0...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-console: 48.0.1 → 49.rc

https://gitlab.gnome.org/GNOME/console/-/compare/48.0.1...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>
Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-tweaks: 46.1 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-tweaks/-/compare/46.1...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
orca: 48.6 → 49.beta

https://gitlab.gnome.org/GNOME/orca/-/compare/48.6...49.beta

Co-authored-by: Bobby Rong <rjl931189261@126.com>
Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-contacts: 48.0 → 49.beta

https://gitlab.gnome.org/GNOME/gnome-contacts/-/compare/48.0...49.beta

Co-authored-by: Bobby Rong <rjl931189261@126.com>
Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
yelp: 42.3 → 49.rc

https://gitlab.gnome.org/GNOME/yelp/-/compare/42.3...49.rc

Co-authored-by: Bobby Rong <rjl931189261@126.com>
Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-boxes: 48.0 → 49.rc.1

https://gitlab.gnome.org/GNOME/gnome-boxes/-/compare/48.0...49.rc.1

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-text-editor: 48.3 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-text-editor/-/compare/48.3...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>
Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-calendar: 48.1 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-calendar/-/compare/48.1...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>
Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-backgrounds: 48.2.1 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-backgrounds/-/compare/48.2.1...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
adwaita-fonts: 48.2 → 49.0

https://gitlab.gnome.org/GNOME/adwaita-fonts/-/compare/48.2...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>
Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
evolution: 3.56.2 → 3.57.3

https://gitlab.gnome.org/GNOME/evolution/-/compare/3.56.2...3.57.3

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gtk-frdp: 0-unstable-2025-03-14 → 0-unstable-2025-08-15

https://gitlab.gnome.org/GNOME/gtk-frdp/-/compare/a0187fa02e1ff249e9583e8c09a2c2f5915ce2a3...b59dc88624511311576dca607d3cb9317569de34

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
libpeas2: 2.0.7 → 2.1.0

https://gitlab.gnome.org/GNOME/libpeas/-/compare/2.0.7...2.1.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
template-glib: 3.36.3 → 3.37.1

https://gitlab.gnome.org/GNOME/template-glib/-/compare/3.36.3...3.37.1

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
glycin-loaders: 1.2.3 → 2.0.rc

New ABI version.

https://gitlab.gnome.org/GNOME/glycin/-/compare/1.2.3...2.0.rc

Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gtkmm4: 4.18.0 → 4.19.1

https://gitlab.gnome.org/GNOME/gtkmm/-/compare/4.18.0...4.19.1

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
libshumate: 1.4.0 → 1.5.beta

https://gitlab.gnome.org/GNOME/libshumate/-/compare/1.4.0...1.5.beta

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
nautilus: 48.3 → 49.rc

https://gitlab.gnome.org/GNOME/nautilus/-/compare/48.3...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-remote-desktop: 48.1 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/compare/48.1...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-shell-extensions: 48.3 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-shell-extensions/-/compare/48.3...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-shell: 48.4 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-shell/-/compare/48.4...49.rc

Co-authored-by: Bobby Rong <rjl931189261@126.com>
Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
vte: 0.80.3 → 0.81.90

https://gitlab.gnome.org/GNOME/vte/-/compare/0.80.3...0.81.90

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>
Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gdm: 48.0 → 49.rc

https://gitlab.gnome.org/GNOME/gdm/-/compare/48.0...49.rc

Co-authored-by: Bobby Rong <rjl931189261@126.com>
Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
mutter: 48.4 → 49.rc

https://gitlab.gnome.org/GNOME/mutter/-/compare/48.4...49.rc

Co-authored-by: Bobby Rong <rjl931189261@126.com>
Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-tecla: 48.0.2 → 49.rc

https://gitlab.gnome.org/GNOME/tecla/-/compare/48.0.2...49.rc

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
localsearch: 3.9.0 → 3.10.rc

https://gitlab.gnome.org/GNOME/localsearch/-/compare/3.9.0...3.10.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gom: 0.5.3 → 0.5.4

https://gitlab.gnome.org/GNOME/gom/-/compare/0.5.3...0.5.4

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
evolution-data-server: 3.56.2 → 3.57.3

https://gitlab.gnome.org/GNOME/evolution-data-server/-/compare/3.56.2...3.57.3

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gtksourceview5: 5.16.0 → 5.17.1

https://gitlab.gnome.org/GNOME/gtksourceview/-/compare/5.16.0...5.17.1

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-online-accounts: 3.54.5 → 3.55.2

https://gitlab.gnome.org/GNOME/gnome-online-accounts/-/compare/3.54.5...3.55.2

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
yelp-xsl: 42.4 → 49.beta

https://gitlab.gnome.org/GNOME/yelp-xsl/-/compare/42.4...49.beta

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
glibmm_2_68: 2.84.0 → 2.85.0

https://gitlab.gnome.org/GNOME/glibmm/-/compare/2.84.0...2.85.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
sysprof: 48.0 → 49.rc

https://gitlab.gnome.org/GNOME/sysprof/-/compare/48.0...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-settings-daemon: 48.1 → 49.rc

https://gitlab.gnome.org/GNOME/gnome-settings-daemon/-/compare/48.1...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
zenity: 4.1.90 → 4.1.99

https://gitlab.gnome.org/GNOME/zenity/-/compare/4.1.90...4.1.99

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
libpanel: 1.10.0 → 1.10.1

https://gitlab.gnome.org/GNOME/libpanel/-/compare/1.10.0...1.10.1

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
libdex: 0.10.1 → 1.0.0

https://gitlab.gnome.org/GNOME/libdex/-/compare/0.10.1...1.0.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-desktop: 44.3 → 44.4

https://gitlab.gnome.org/GNOME/gnome-desktop/-/compare/44.3...44.4

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
libadwaita: 1.7.8 → 1.8.rc

https://gitlab.gnome.org/GNOME/libadwaita/-/compare/1.7.8...1.8.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gjs: 1.84.2 → 1.85.90

https://gitlab.gnome.org/GNOME/gjs/-/compare/1.84.2...1.85.90

Co-authored-by: dish <pyrox@pyrox.dev>
Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
adwaita-icon-theme: 48.0 → 49.rc

https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/compare/48.0...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gtk4: 4.18.6 → 4.20.0

https://gitlab.gnome.org/GNOME/gtk/-/compare/4.18.6...4.20.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gtk3: 3.24.49 → 3.24.50

https://gitlab.gnome.org/GNOME/gtk/-/compare/3.24.49...3.24.50

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
at-spi2-core: 2.56.2 → 2.57.2

https://gitlab.gnome.org/GNOME/at-spi2-core/-/compare/2.56.2...2.57.2

Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gsettings-desktop-schemas: 48.0 → 49.rc

https://gitlab.gnome.org/GNOME/gsettings-desktop-schemas/-/compare/48.0...49.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gdk-pixbuf: 2.42.12 → 2.43.5

https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/compare/2.42.12...2.43.5

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
pango: 1.56.4 → 1.57.0

https://gitlab.gnome.org/GNOME/pango/-/compare/1.56.4...1.57.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
glib: 2.84.4 → 2.86.0

https://gitlab.gnome.org/GNOME/glib/-/compare/2.84.4...2.86.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gi-docgen: 2025.3 → 2025.4

https://gitlab.gnome.org/GNOME/gi-docgen/-/compare/2025.3...2025.4

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
rygel: 0.44.2 → 45.rc

https://gitlab.gnome.org/GNOME/rygel/-/compare/rygel-0.44.2...45.rc

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
libglycin: 1.2.3 → 2.0.rc

New ABI version.

https://gitlab.gnome.org/GNOME/glycin/-/compare/1.2.3...2.0.rc

Keep the old patch for programs that still depend on glycin 2.0 crate.

Co-authored-by: Bobby Rong <rjl931189261@126.com>

bobby285271 committed:
nixos/tests: gnome-session-binary → gnome-session-service

https://github.com/GNOME/gnome-session/commit/cb85168d2694e9c36d00ce1407882c88fadd4f59

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/gnome-49-in-25-05/68589/2

bobby285271 committed:
gnome-session-ctl: 47.0.1 → 49.0

https://github.com/nix-community/gnome-session-ctl/compare/47.0.1...49.0

bobby285271 committed:
budgie-control-center: 1.4.0 → 1.4.1

https://github.com/BuddiesOfBudgie/budgie-control-center/compare/v1.4.0...v1.4.1

Adjusted to GSD 49.

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

bobby285271 committed:
budgie-desktop: 10.9.2 → 10.9.3

https://github.com/BuddiesOfBudgie/budgie-desktop/compare/v10.9.2...v10.9.3

Adjusted to GSD 49.

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

bobby285271 committed:
nixos/tests/budgie: Adjust to Budgie 10.9.3

Restore the wait_for_x test since our XDG_CURRENT_DESKTOP changed.
See the `session_is_systemd_aware` part in
nixos/modules/services/x11/display-managers/default.nix.

MisileLab commented:
gdk-pixbuf 2.44.2 released, by the way.
https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/compare/2.43.5...2.44.2?from_project_id=1548

jtojnar committed:
nixos/gnome-initial-setup: Create group

User is now created dynamically by GDM (though we did not create it ourselves previously):

https://github.com/GNOME/gnome-initial-setup/commit/3ced516c03c5267d0b11dc8649e837f05fc8bfa3

jtojnar committed:
nixos/gnome-initial-setup: Remove gnome-welcome-tour service dependency

This has been dropped in GNOME 40:

https://github.com/GNOME/gnome-initial-setup/commit/2bf1bcaa81c8733665cc33be78652969dd07cd9e

jtojnar committed:
nixos/gnome-initial-setup: Adjust service dependencies

`gnome-initial-setup-copy-worker.service` was moved from `gnome-session.target.wants` to `basic.target.wants/` in GNOME 46.1:

https://github.com/GNOME/gnome-initial-setup/commit/34680e68e049c83eda8ea4dbd0ea3f2fbbf00c89

And now GNOME 49 moved it again to `graphical-session-pre.target.wants/`:

https://github.com/GNOME/gnome-initial-setup/commit/40c3de824c13f240e27567307b0ebaf4363e4f42

jtojnar committed:
nixos/gnome: Install GNOME Software using module

gnome-software now uses systemd instead of autostart so we need to add it to `systemd.packages`:
https://github.com/GNOME/gnome-software/commit/08805775079ef6190792f254b799cd920f248d7c

jtojnar committed:
nixos/orca: Remove from display-manager path

GDM no longer includes `orca-autostart.desktop`:
https://github.com/GNOME/gdm/commit/cb0a3d4deb0ba01cf4277621f2b8ccfa527b4af3

jtojnar committed:
gnome-characters: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-characters/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-terminal: 3.56.2 → 3.58.0

https://gitlab.gnome.org/GNOME/gnome-terminal/-/compare/3.56.2...3.58.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-initial-setup: 49.beta.1 → 49.0

https://gitlab.gnome.org/GNOME/gnome-initial-setup/-/compare/49.beta.1...49.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-mahjongg: 49.rc → 49.0.1

https://gitlab.gnome.org/GNOME/gnome-mahjongg/-/compare/49.rc...49.0.1

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
d-spy: 48.0 → 49.1

https://gitlab.gnome.org/GNOME/d-spy/-/compare/48.0...49.1

Co-authored-by: Bobby Rong <rjl931189261@126.com>
Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-font-viewer: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-font-viewer/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-calendar: 49.rc → 49.0.1

https://gitlab.gnome.org/GNOME/gnome-calendar/-/compare/49.rc...49.0.1

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-console: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/console/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
xdg-desktop-portal-gnome: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-logs: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-logs/-/compare/49.rc...49.0

Co-authored-by: Bobby Rong <rjl931189261@126.com>
Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-applets: 3.56.0 → 3.58.0

https://gitlab.gnome.org/GNOME/gnome-applets/-/compare/3.56.0...3.58.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-chess: 49.0 → 49.2

https://gitlab.gnome.org/GNOME/gnome-chess/-/compare/49.0...49.2

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-software: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-software/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
decibels: 48.0 → 49.0

https://gitlab.gnome.org/GNOME/decibels/-/compare/48.0...49.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-calculator: 49.rc → 49.0.1

https://gitlab.gnome.org/GNOME/gnome-calculator/-/compare/49.rc...49.0.1

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-text-editor: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-text-editor/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
simple-scan: 49.0 → 49.0.1

https://gitlab.gnome.org/GNOME/simple-scan/-/compare/49.0...49.0.1

Co-authored-by: Bobby Rong <rjl931189261@126.com>
Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
orca: 49.beta → 49.1

https://gitlab.gnome.org/GNOME/orca/-/compare/49.beta...49.1

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-tour: 48.1 → 49.0

https://gitlab.gnome.org/GNOME/gnome-tour/-/compare/48.1...49.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-contacts: 49.beta → 49.0

https://gitlab.gnome.org/GNOME/gnome-contacts/-/compare/49.beta...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-boxes: 49.rc.1 → 49.0

https://gitlab.gnome.org/GNOME/gnome-boxes/-/compare/49.rc.1...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-builder: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-builder/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
loupe: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/loupe/-/compare/49.rc...49.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-control-center: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-control-center/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-weather: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-weather/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
baobab: 48.0 → 49.0

https://gitlab.gnome.org/GNOME/baobab/-/compare/48.0...49.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-connections: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-connections/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-tweaks: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-tweaks/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-system-monitor: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-system-monitor/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
rygel: 45.rc → 45.0

https://gitlab.gnome.org/GNOME/rygel/-/compare/45.rc...45.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
snapshot: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/snapshot/-/compare/49.rc...49.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-mines: 49.rc → 49.0.1

https://gitlab.gnome.org/GNOME/gnome-mines/-/compare/49.rc...49.0.1

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
nautilus-python: 4.0.1 → 4.1.0

https://gitlab.gnome.org/GNOME/nautilus-python/-/compare/4.0.1...4.1.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-firmware: 47.0 → 49.0

https://gitlab.gnome.org/World/gnome-firmware/-/compare/47.0...49.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-maps: 49.rc → 49.1

https://gitlab.gnome.org/GNOME/gnome-maps/-/compare/49.rc...49.1

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
yelp: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/yelp/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
metacity: 3.57.1 → 3.58.1

https://gitlab.gnome.org/GNOME/metacity/-/compare/3.57.1...3.58.1

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-session: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-session/-/compare/49.rc...49.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-music: 49.rc → 49.0.2

https://gitlab.gnome.org/GNOME/gnome-music/-/compare/49.rc...49.0.2

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
epiphany: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/epiphany/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
quadrapassel: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/quadrapassel/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-sudoku: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-sudoku/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-flashback: 3.56.0 → 3.58.0

https://gitlab.gnome.org/GNOME/gnome-flashback/-/compare/3.56.0...3.58.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-nibbles: 4.4.rc2 → 4.4.1

https://gitlab.gnome.org/GNOME/gnome-nibbles/-/compare/4.4.rc2...4.4.1

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
dconf-editor: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/dconf-editor/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-tetravex: 3.38.2 → 3.38.3

https://gitlab.gnome.org/GNOME/gnome-tetravex/-/compare/3.38.2...3.38.3

Co-authored-by: Bobby Rong <rjl931189261@126.com>
Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-backgrounds: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-backgrounds/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-user-docs: 48.2 → 49.0

https://gitlab.gnome.org/GNOME/gnome-user-docs/-/compare/48.2...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gucharmap: 16.0.2 → 17.0.0

https://gitlab.gnome.org/GNOME/gucharmap/-/compare/16.0.2...17.0.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
libpeas2: 2.1.0 → 2.2.0

https://gitlab.gnome.org/GNOME/libpeas/-/compare/2.1.0...2.2.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
template-glib: 3.37.1 → 3.38.0

https://gitlab.gnome.org/GNOME/template-glib/-/compare/3.37.1...3.38.0

Co-authored-by: Bobby Rong <rjl931189261@126.com>
Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-remote-desktop: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-shell-extensions: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-shell-extensions/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
glycin-loaders: 2.0.rc → 2.0.2

https://gitlab.gnome.org/GNOME/glycin/-/compare/2.0.rc...2.0.2

Co-authored-by: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gtkmm4: 4.19.1 → 4.20.0

https://gitlab.gnome.org/GNOME/gtkmm/-/compare/4.19.1...4.20.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
libshumate: 1.5.beta → 1.5.0.1

https://gitlab.gnome.org/GNOME/libshumate/-/compare/1.5.beta...1.5.0.1

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
evolution: 3.57.3 → 3.58.0

https://gitlab.gnome.org/GNOME/evolution/-/compare/3.57.3...3.58.0

jtojnar committed:
gnome-shell: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-shell/-/compare/49.rc...49.0

jtojnar committed:
nautilus: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/nautilus/-/compare/49.rc...49.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-panel: 3.56.0 → 3.58.1

https://gitlab.gnome.org/GNOME/gnome-panel/-/compare/3.56.0...3.58.1

Co-authored-by: Bobby Rong <rjl931189261@126.com>
Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gtksourceview5: 5.17.1 → 5.18.0

https://gitlab.gnome.org/GNOME/gtksourceview/-/compare/5.17.1...5.18.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
vte: 0.81.90 → 0.82.0

https://gitlab.gnome.org/GNOME/vte/-/compare/0.81.90...0.82.0

Co-authored-by: Bobby Rong <rjl931189261@126.com>
Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-clocks: 48.0 → 49.0

https://gitlab.gnome.org/GNOME/gnome-clocks/-/compare/48.0...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-tecla: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/tecla/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
mutter: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/mutter/-/compare/49.rc...49.0

jtojnar committed:
localsearch: 3.10.rc → 3.10.0

https://gitlab.gnome.org/GNOME/localsearch/-/compare/3.10.rc...3.10.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gdm: 49.rc → 49.0.1

https://gitlab.gnome.org/GNOME/gdm/-/compare/49.rc...49.0.1

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
evolution-data-server: 3.57.3 → 3.58.0

https://gitlab.gnome.org/GNOME/evolution-data-server/-/compare/3.57.3...3.58.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
gnome-online-accounts: 3.55.2 → 3.56.0

https://gitlab.gnome.org/GNOME/gnome-online-accounts/-/compare/3.55.2...3.56.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
glibmm_2_68: 2.85.0 → 2.86.0

https://gitlab.gnome.org/GNOME/glibmm/-/compare/2.85.0...2.86.0

Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

jtojnar committed:
yelp-xsl: 49.beta → 49.0

https://gitlab.gnome.org/GNOME/yelp-xsl/-/compare/49.beta...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gnome-settings-daemon: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gnome-settings-daemon/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
sysprof: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/sysprof/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
zenity: 4.1.99 → 4.2.0

https://gitlab.gnome.org/GNOME/zenity/-/compare/4.1.99...4.2.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
libglycin: 2.0.rc → 2.0.2

https://gitlab.gnome.org/GNOME/glycin/-/compare/2.0.rc...2.0.2

jtojnar committed:
gvfs: 1.57.2 → 1.58.0

https://gitlab.gnome.org/GNOME/gvfs/-/compare/1.57.2...1.58.0

jtojnar committed:
libpanel: 1.10.1 → 1.10.2

https://gitlab.gnome.org/GNOME/libpanel/-/compare/1.10.1...1.10.2

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
libadwaita: 1.8.rc → 1.8.0

https://gitlab.gnome.org/GNOME/libadwaita/-/compare/1.8.rc...1.8.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gjs: 1.85.90 → 1.86.0

https://gitlab.gnome.org/GNOME/gjs/-/compare/1.85.90...1.86.0

jtojnar committed:
adwaita-icon-theme: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/compare/49.rc...49.0

Changelog-Reviewed-By: Bobby Rong <rjl931189261@126.com>

jtojnar committed:
gtk4: 4.20.0 → 4.20.2

https://gitlab.gnome.org/GNOME/gtk/-/compare/4.20.0...4.20.2

jtojnar committed:
gtk3: 3.24.50 → 3.24.51

https://gitlab.gnome.org/GNOME/gtk/-/compare/3.24.50...3.24.51

jtojnar committed:
gdk-pixbuf: 2.43.5 → 2.44.3

https://gitlab.gnome.org/GNOME/gdk-pixbuf/-/compare/2.43.5...2.44.3

jtojnar committed:
at-spi2-core: 2.57.2 → 2.58.0

https://gitlab.gnome.org/GNOME/at-spi2-core/-/compare/2.57.2...2.58.0

jtojnar committed:
gsettings-desktop-schemas: 49.rc → 49.0

https://gitlab.gnome.org/GNOME/gsettings-desktop-schemas/-/compare/49.rc...49.0

jtojnar committed:
gobject-introspection: 1.84.0 → 1.86.0

https://gitlab.gnome.org/GNOME/gobject-introspection/-/compare/1.84.0...1.86.0

jtojnar committed:
gtk-doc: 1.34.0 → 1.35.1

https://gitlab.gnome.org/GNOME/gtk-doc/-/compare/1.34.0...1.35.1

jtojnar commented:
Hmm, that would be good. I did not realise it does not have gnome team as maintainer so it got skipped.

bobby285271 committed:
python3.pkgs.pygobject3: 3.50.0 → 3.54.3

https://gitlab.gnome.org/GNOME/pygobject/-/compare/3.50.0...3.54.3

bobby285271 committed:
libpeas: Switch to girepository-2.0

https://gitlab.gnome.org/GNOME/libpeas/-/issues/58

bobby285271 committed:
gitg: Switch to girepository-2.0

See https://gitlab.gnome.org/GNOME/libpeas/-/issues/58.

Fix building with libpeas1 (ported to girepository-2.0).

bobby285271 committed:
rhythmbox: Switch to girepository-2.0

https://gitlab.gnome.org/GNOME/rhythmbox/-/issues/2113

bobby285271 committed:
python3.pkgs.gst-python: Disable TestNotInitialized test

https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/4322

lesbiannix committed:
networkmanager: 1.52.1 → 1.54.1

https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/compare/1.52.1...1.54.1

Picked from Nixpkgs pull request 446121. Fixes building with PyGObject 3.52+.

Co-authored-by: Martin Weinelt <mweinelt@users.noreply.github.com>

bobby285271 committed:
totem: Switch to girepository-2.0

See https://gitlab.gnome.org/GNOME/libpeas/-/issues/58.

bobby285271 committed:
endeavour: Switch to girepository-2.0

See https://gitlab.gnome.org/GNOME/libpeas/-/issues/58.

bobby285271 commented:
I think I want an ACK on f74e2242efe036afc12f1760a0a82e609cec6cec :joy: . The commit is [unreleased upstream](https://github.com/GNOME/libpeas/commit/73e25b6059d2fdc090a3feb8341ff902c3ec0d16) and it looks like only Fedora [lands that](https://src.fedoraproject.org/rpms/libpeas1/blob/rawhide/f/73e25b6059d2fdc090a3feb8341ff902c3ec0d16.patch) and Arch just [disabled](https://gitlab.archlinux.org/archlinux/packaging/packages/pygobject/-/issues/3#note_256714) python3 loader. It looks like we mostly need to visit every libpeas1 revdeps, though I think most of the fix are either [trivial](https://github.com/NixOS/nixpkgs/commit/b060aab050be4b2f37d7585d19150f740dc047e3) or already fixed somewhere:

<details>


- [ ] pkgs/applications/audio/pragha/default.nix
- [ ] pkgs/applications/networking/mailreaders/astroid/default.nix
- [ ] pkgs/by-name/bu/budgie-analogue-clock-applet/package.nix
- [ ] pkgs/by-name/bu/budgie-desktop/package.nix
- [ ] pkgs/by-name/bu/budgie-user-indicator-redux/package.nix
- [ ] pkgs/by-name/bu/budgie-systemmonitor-applet/package.nix
- [ ] pkgs/by-name/di/diodon/package.nix
- [x] pkgs/by-name/en/endeavour/package.nix
- [ ] pkgs/by-name/en/entangle/package.nix
- [ ] pkgs/by-name/eo/eog/package.nix
- [ ] pkgs/by-name/ge/geary/package.nix
- [ ] pkgs/by-name/ge/gedit/package.nix
- [x] pkgs/by-name/gi/gitg/package.nix
- [ ] pkgs/by-name/gn/gnome-pomodoro/package.nix
- [ ] pkgs/by-name/gt/gtkd/package.nix
- [ ] pkgs/by-name/mi/microdnf/package.nix
- [ ] pkgs/by-name/pi/pitivi/package.nix
- [x] pkgs/by-name/rh/rhythmbox/package.nix
- [x] pkgs/by-name/to/totem/package.nix
- [ ] pkgs/by-name/xe/xed-editor/package.nix
- [ ] pkgs/by-name/xv/xviewer/package.nix
- [ ] pkgs/desktops/mate/eom/default.nix
- [ ] pkgs/desktops/mate/pluma/default.nix


</details>

bobby285271 committed:
xed-editor: Switch to girepository-2.0

See https://gitlab.gnome.org/GNOME/libpeas/-/issues/58.

bobby285271 committed:
xviewer: Switch to girepository-2.0

See https://gitlab.gnome.org/GNOME/libpeas/-/issues/58.

bobby285271 committed:
mate.eom: Switch to girepository-2.0

See https://gitlab.gnome.org/GNOME/libpeas/-/issues/58.

bobby285271 committed:
mate.pluma: Switch to girepository-2.0

See https://gitlab.gnome.org/GNOME/libpeas/-/issues/58.

bobby285271 committed:
eog: Switch to girepository-2.0

See https://gitlab.gnome.org/GNOME/libpeas/-/issues/58.

bobby285271 committed:
budgie-desktop: Switch to libpeas-2

See https://gitlab.gnome.org/GNOME/libpeas/-/issues/58.

bobby285271 committed:
budgie-analogue-clock-applet: Switch to libpeas-2

See https://gitlab.gnome.org/GNOME/libpeas/-/issues/58.

bobby285271 committed:
budgie-systemmonitor-applet: Switch to libpeas-2

See https://gitlab.gnome.org/GNOME/libpeas/-/issues/58.

bobby285271 committed:
budgie-user-indicator-redux: Switch to libpeas-2

See https://gitlab.gnome.org/GNOME/libpeas/-/issues/58.

bobby285271 committed:
budgie-media-player-applet: Switch to libpeas-2

See https://gitlab.gnome.org/GNOME/libpeas/-/issues/58.

bobby285271 committed:
astroid: Switch to girepository-2.0

See https://gitlab.gnome.org/GNOME/libpeas/-/issues/58.

bobby285271 committed:
entangle: Switch to girepository-2.0

See https://gitlab.gnome.org/GNOME/libpeas/-/issues/58.

https://github.com/GNOME/glib/commit/f55b18e11e88bd58f20ce9d7f3c29ccf9e17514e

bobby285271 committed:
pantheon.elementary-mail: Fix build with evolution-data-server 3.57.1

The evolution-data-server 3.57.1 release contains API changes in libcamel,
which are incompatible with the previous versions.

bobby285271 committed:
unihan-database: 15.1.0 → 17.0.0

https://www.unicode.org/versions/Unicode17.0.0/
Updated together with gucharmap.

bobby285271 committed:
unicode-character-database: 16.0.0 → 17.0.0

https://www.unicode.org/versions/Unicode17.0.0/
Updated together with gucharmap.

vcunat commented:
We should have a new `staging-next` cycle in a few days, and I wonder &ndash; maybe it would be good to have this GNOME upgrade in it already (and continue this fixing during `staging-next`).  Otherwise we risk it getting into nixpkgs master only a couple days before releasing NixOS 25.11, which I'd find not great.

EDIT: or get at least the mass-rebuild parts into that staging-next iteration, as currently you have
> Rebuild: linux 48589, darwin 31935

jtojnar commented:
It still remains blocked on gdm crashing because of the dynamic user changes:

https://github.com/NixOS/nixpkgs/pull/440720#issuecomment-3349801207

I am looking into it now but not sure if it will make it on time.

jtojnar committed:
glib: 2.86.0 → 2.86.1

https://gitlab.gnome.org/GNOME/glib/-/compare/2.86.0...2.86.1

jtojnar commented:
I have resolved the blocking issue by hardcoding greeter users rather than trying to figure out how to make systemd-userdb work, GDM now starts fine.

All pre-installed apps appear to launch.

It is not the most up to date version and there are still some [minor tasks](https://github.com/orgs/NixOS/projects/101) but that can be done on master.

Rebased and updated glib (untested other than that it itself builds) to fix https://github.com/NixOS/nixpkgs/issues/457619.

The eval failure seems to come from base branch.

jtojnar committed:
nixos/rl-2511: Add GNOME 49 notes

leona-ya reviewed (approved):
I had a rough look, seems fine to me. I'd still appreciate another review by a person more experienced with GNOME, but otherwise I'd also merge this in 5-16 hours

valyntyler commented:
Sorry for the dumb question but when do we expect this to be in unstable?

vcunat commented:
This PR should land this week in `nixos-unstable`.

andersk commented:
Please update `gnomeExtensions` too:

- #461080

peigongdsd commented:
> > > gnome 49 apps are extremely buggy why?
> > 
> > If they are, you should be opening an issue on nixpkgs for the application(s) with the bugs.
> 
> I was having a graphics issue that crashed my whole DE randomly. Apparently it's tied to lack of "native" support for Vulkan, which GNOME 49 is now asking for. I'm not going to open an issue because I know too little about this and I have too little time. I've fixed it, for now, with `home-manager.options.nixGL.vulkan.enable = true`.
> 
> update: it doesn't even fix all the issues. I give up.
> update update: as far as I can tell the issue was actually related to opengl drivers. `mesa` package version 2025.3.0 from unstable seems to be broken. Setting `hardware.graphics.package` to `pkgs.mesa` version 25.0.7 from `nixpkgs-25.05` seems to solve the issue.


I'm currently also facing problems on gtk4 apps with vulkan backend. Use GSK_BACKEND=ngl to switch the renderer to opengl fixed the issue.

vcunat commented:
I think these are solved by https://github.com/NixOS/nixpkgs/pull/463695 which is contained in #461523
